pkg/resources/pods.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"knative.dev/serving/pkg/apis/serving"
+)
+
+type scopedPodCounter struct {
+	podsLister  corev1listers.PodLister
+	namespace   string
+	serviceName string
+	pods        []*corev1.Pod
+}
+
+// NotRunningPodCounter provides a count of pods currently not in the
+// RUNNING state. The interface prevents the users from needing to
+// know how counts are performed.
+// The first int return value represents the number of pending pods
+// The second int return value represents the number of terminating pods
+// The error value is returned if the NotRunningCount is unable to
+// calculate a value.
+type NotRunningPodCounter interface {
+	Get() error
+	TerminatingCount() int
+	PendingCount() int
+}
+
+// NewScopedPodsCounter creates a NotRunningPodCounter that uses
+// a count of pods for a namespace/serviceName as the value
+// of not running pods. The values returned by NotRunningCount() will vary
+// over time.
+// lister is used to retrieve pods for counting with the
+// scope of namespace/serviceName.
+func NewScopedPodsCounter(lister corev1listers.PodLister, namespace, serviceName string) NotRunningPodCounter {
```

Comment:
It is a bit strange that `NewScopedPodsCounter` returns `NotRunningPodCounter`. I'd expect `NewNotRunningPodCounter` to be the name?